### PR TITLE
[Storage][Sharding] Prepare to split state kvs to a separate db.

### PR DESCRIPTION
**Pull request by grao1991**
### Description

Configure the new db. No real behavior change for now.

### CODE REVIEW

File: storage/aptosdb/src/db_options.rs [hunk 1]
```diff
@@ -86,6 +94,31 @@ pub(super) fn gen_state_merkle_cfds(rocksdb_config: &RocksdbConfig) -> Vec<Colum
     cfds
 }
 
+pub(super) fn gen_kv_cfds(rocksdb_config: &RocksdbConfig) -> Vec<ColumnFamilyDescriptor> {
```
Comment (msmouse) [inline storage/aptosdb/src/db_options.rs:97]: So much duplication.. let's do some house keeping?

File: storage/aptosdb/src/lib.rs [hunk 2]
```diff
@@ -259,13 +261,15 @@ impl AptosDB {
...
         state_merkle_rocksdb: DB,
+        kv_rocksdb: DB,
         pruner_config: PrunerConfig,
         buffered_state_target_items: usize,
         max_nodes_per_lru_cache_shard: usize,
         hack_for_tests: bool,
     ) -> Self {
         let arc_ledger_rocksdb = Arc::new(ledger_rocksdb);
         let arc_state_merkle_rocksdb = Arc::new(state_merkle_rocksdb);
+        let arc_kv_rocksdb = Arc::new(kv_rocksdb);
```
Comment (msmouse) [inline storage/aptosdb/src/lib.rs:272]: Let's make it optional and configurable? (on a node in production don't create the DB folder and allocate db threads)

File: storage/aptosdb/src/db_options.rs (hunk 1 above)
Comment (grao1991) [reply storage/aptosdb/src/db_options.rs:97]: Done

File: storage/aptosdb/src/lib.rs (hunk 2 above)
Comment (grao1991) [reply storage/aptosdb/src/lib.rs:272]: Done

**Review by zjma — APPROVED** [review]

File: storage/aptosdb/src/lib.rs [hunk 3]
```diff
@@ -245,6 +246,7 @@ impl Drop for RocksdbPropertyReporter {
 pub struct AptosDB {
     ledger_db: Arc<DB>,
     state_merkle_db: Arc<DB>,
+    _kv_db: Arc<Option<DB>>,
```
Comment (msmouse) [inline storage/aptosdb/src/lib.rs:249]: nit: `Arc<Option<DB>>` is a bit weird vs `Option<Arc<DB>>` to me, rationale?

File: config/src/config/storage_config.rs [hunk 4]
```diff
@@ -53,6 +53,8 @@ impl Default for RocksdbConfig {
 pub struct RocksdbConfigs {
     pub ledger_db_config: RocksdbConfig,
     pub state_merkle_db_config: RocksdbConfig,
+    pub use_kv_db: bool,
+    pub kv_db_config: RocksdbConfig,
```
Comment (msmouse) [inline config/src/config/storage_config.rs:57]: nit: `state_kv_db` is more symmetric  to `state_merkle_db`

**Review by msmouse — APPROVED** [review]

**Pull request merged**